classes/context/SubEditorsDAO.php

```diff
@@ -66,29 +83,18 @@ public function deleteEditor($contextId, $assocId, $userId, $assocType)
     /**
      * Retrieve a list of all sub editors assigned to the specified submission group.
      *
-     * @param int $assocId
+     * @param int[] $assocIds Section or category ids
      * @param int $assocType ASSOC_TYPE_SECTION or ASSOC_TYPE_CATEGORY
-     * @param int $contextId
      *
-     * @return array matching Users
+     * @return Collection result rows with user_id and user_group_id columns
      */
-    public function getBySubmissionGroupId($assocId, $assocType, $contextId)
+    public function getBySubmissionGroupIds(array $assocIds, int $assocType, int $contextId): Collection
     {
-        $userDao = Repo::user()->dao;
-        $result = $this->retrieve(
-            'SELECT	u.*
-			FROM	subeditor_submission_group e
-				JOIN users u ON (e.user_id = u.user_id)
-			WHERE	e.context_id = ? AND
-				e.assoc_id = ? AND e.assoc_type = ?',
-            [(int) $contextId, (int) $assocId, (int) $assocType]
-        );
-
-        $users = [];
-        foreach ($result as $row) {
-            $users[$row->user_id] = $userDao->fromRow($row);
-        }
-        return $users;
+        return DB::table('subeditor_submission_group')
+            ->where('assoc_type', '=', $assocType)
+            ->where('context_id', '=', $contextId)
+            ->whereIn('assoc_id', $assocIds)
+            ->get(['user_id', 'user_group_id']);
```

Comment:
This is probably the first place in the system where we're exposing `database_column_naming` outside of a DAO's internal implementation rather than our usual convention of `camelCase` in PHP. I know it adds cruft but I'm pretty hesitant to break our current clean separation of styles. I'd suggest giving these lowerCamelCase aliases if this is going to be a public function.

Reply:
Good call. I've aliased the select columns as `userId` and `userGroupId`.